lib/core/sync_job_runner.rb

```diff
@@ -33,13 +34,14 @@ def initialize(connector_settings, service_type)
 
     def execute
       unless @connector_settings.configuration_initialized?
-        @connector_settings.update_configuration(@connector_instance.configurable_fields)
+        @connector_settings.initialize_configuration(@connector_instance.configurable_fields)
       end
 
       validate_configuration!
       return unless should_sync?
```

Comment:
Should this (or `should_sync?`) also check to ensure that the current status is not `needs_configuration`?

Reply:
Good call, though I need to double check it with @sphilipse.

So connector says that it `needs_configuration`, what will be the status of connector after user updates the config in Kibana?

Reply:
Status will be `configured`